src/OpenLoco/src/Map/Track/TrackData.cpp

```diff
@@ -389,8 +392,602 @@ namespace OpenLoco::World::TrackData
+        "MiscFlags::steepSlope",
+        "MiscFlags::curveSlope",
+        "MiscFlags::diagonal",
+        "MiscFlags::verySmallCurve",
+        "MiscFlags::smallCurve",
+        "MiscFlags::curve",
+        "MiscFlags::largeCurve",
+        "MiscFlags::sBendCurve",
+        "MiscFlags::oneSided",
+    };
+
+    std::string toMiscString(uint16_t val)
+    {
+        std::string out;
+        for (auto i = 0; i < 10; ++i)
+        {
+            if (val & (1U << i))
+            {
+                if (out.empty())
+                {
+                    out = _miscToString[i];
+                }
+                else
+                {
+                    out += " | " + _miscToString[i];
+                }
+            }
+        }
+        if (out.empty())
+        {
+            out = "MiscFlags::none";
+        }
+        return out;
+    }
+    std::string toTrackPieceString(uint16_t val)
+    {
+        std::string out;
+        for (auto i = 0; i < 11; ++i)
+        {
+            if (val & (1U << i))
+            {
+                if (out.empty())
+                {
+                    out = _pieceFlagsToString[i];
+                }
+                else
+                {
+                    out += " | " + _pieceFlagsToString[i];
+                }
+            }
+        }
+        if (out.empty())
+        {
+            out = "TrackPieceFlags::none";
+        }
+        return out;
+    }
+
+    const MiscData& getTrackMiscData(size_t trackId)
+    {
+        return _miscData[trackId];
+    }
+
     uint16_t getTrackCompatibleFlags(size_t trackId)
     {
+        for (auto i = 0; i < 44; ++i)
+        {
+            MiscData data{};
+            data.costFactor = _costFactor[i];
+            data.flags = static_cast<MiscFlags>(_flags[i]);
+            data.reverseTrackId = _variousThings[i * 8 + 0];
+            data.reverseRotation = _variousThings[i * 8 + 1];
+            data.signalHeightOffsetLeft = _variousThings[i * 8 + 2];
+            data.signalHeightOffsetRight = _variousThings[i * 8 + 3];
+            data.compatibleFlags = static_cast<TrackPieceFlags>(_compatibleFlags[i]);
+            data.curveSpeedFraction = _curveSpeedFraction[i];
+            data.unkWeighting = _unkWeighting[i];
+            data.sparkDirection = _sparkDirection[i];
+
+            fmt::println("MiscData {{");
+            fmt::println(".costFactor = 0x{:03X},\n.flags = {},\n.reverseTrackId = {},", data.costFactor, toMiscString(enumValue(data.flags)), data.reverseTrackId);
+            fmt::println(".reverseRotation = {},\n.signalHeightOffsetLeft = {},\n.signalHeightOffsetRight = {},", data.reverseRotation, data.signalHeightOffsetLeft, data.signalHeightOffsetRight);
+            fmt::println(".compatibleFlags = {},\n.curveSpeedFraction = 0x{:04X},\n.unkWeighting = {},\n.sparkDirection = {}", toTrackPieceString(enumValue(data.compatibleFlags)), data.curveSpeedFraction, data.unkWeighting, data.sparkDirection);
+            fmt::println("}},");
```

Comment:
@LeftofZen Considering these (and draft status), I don't think this is intended for review or indeed merging yet. :D